### PR TITLE
db.describe: add JSON support

### PR DESCRIPTION
**Pull request by kritibirda26**
Add JSON support to db.describe module. The JSON output looks like:

```json
{
    "table": "mysoils",
    "description": "",
    "insert": null,
    "delete": null,
    "ncols": 7,
    "nrows": 1428,
    "columns": [
        {
            "position": 1,
            "column": "cat",
            "description": "",
            "type": "INTEGER",
            "length": 20,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 2,
            "column": "OBJECTID",
            "description": "",
            "type": "INTEGER",
            "length": 20,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 3,
            "column": "AREA",
            "description": "",
            "type": "DOUBLE PRECISION",
            "length": 20,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 4,
            "column": "PERIMETER",
            "description": "",
            "type": "DOUBLE PRECISION",
            "length": 20,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 5,
            "column": "GSLNC250_",
            "description": "",
            "type": "INTEGER",
            "length": 20,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 6,
            "column": "GSLNC250_I",
            "description": "",
            "type": "INTEGER",
            "length": 20,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 7,
            "column": "GSL_NAME",
            "description": "",
            "type": "CHARACTER",
            "length": 6,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        }
    ]
}
```

Where `???` was used as special string for unknown in plain format, I have changed it to null. For instance, select, update, delete and insert privilege information field.

### CODE REVIEW

File: db/db.describe/db.describe.html [hunk 1]
```diff
@@ -61,6 +61,115 @@ <h3>DBF example</h3>
 [...]
 </pre></div>
 
+<h3>JSON Output</h3>
+<div class="code"><pre>
+db.describe mysoils
```
Comment (cwhite911) [inline db/db.describe/db.describe.html:66]:
```suggestion
db.describe table=hospitals format=json
```

File: db/db.describe/db.describe.html [hunk 2]
```diff
@@ -61,6 +61,115 @@ <h3>DBF example</h3>
...
+</pre></div>
+
+<div class="code"><pre>
+{
```
Comment (cwhite911) [inline db/db.describe/db.describe.html:70]:
```suggestion
{
    "table": "hospitals",
    "description": "",
    "insert": null,
    "delete": null,
    "ncols": 16,
    "nrows": 160,
    "columns": [
        {
            "position": 1,
            "column": "cat",
            "description": "",
            "type": "INTEGER",
            "length": 20,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 2,
            "column": "OBJECTID",
            "description": "",
            "type": "INTEGER",
            "length": 20,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 3,
            "column": "AREA",
            "description": "",
            "type": "DOUBLE PRECISION",
            "length": 20,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 4,
            "column": "PERIMETER",
            "description": "",
            "type": "DOUBLE PRECISION",
            "length": 20,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 5,
            "column": "HLS_",
            "description": "",
            "type": "INTEGER",
            "length": 20,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 6,
            "column": "HLS_ID",
            "description": "",
            "type": "INTEGER",
            "length": 20,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 7,
            "column": "NAME",
            "description": "",
            "type": "CHARACTER",
            "length": 45,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 8,
            "column": "ADDRESS",
            "description": "",
            "type": "CHARACTER",
            "length": 35,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 9,
            "column": "CITY",
            "description": "",
            "type": "CHARACTER",
            "length": 16,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 10,
            "column": "ZIP",
            "description": "",
            "type": "CHARACTER",
            "length": 5,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 11,
            "column": "COUNTY",
            "description": "",
            "type": "CHARACTER",
            "length": 12,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 12,
            "column": "PHONE",
            "description": "",
            "type": "CHARACTER",
            "length": 14,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 13,
            "column": "CANCER",
            "description": "",
            "type": "CHARACTER",
            "length": 4,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 14,
            "column": "POLYGONID",
            "description": "",
            "type": "INTEGER",
            "length": 20,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 15,
            "column": "SCALE",
            "description": "",
            "type": "DOUBLE PRECISION",
            "length": 20,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        },
        {
            "position": 16,
            "column": "ANGLE",
            "description": "",
            "type": "DOUBLE PRECISION",
            "length": 20,
            "scale": 0,
            "precision": 0,
            "default": null,
            "nullok": true,
            "select": null,
            "update": null
        }
    ]
}
```

File: db/db.describe/main.c [hunk 3]
```diff
@@ -115,6 +176,9 @@ static void parse_command_line(int argc, char **argv)
     if ((db = db_get_default_database_name()))
         database->answer = (char *)db;
 
+    format_opt = G_define_standard_option(G_OPT_F_FORMAT);
+    format_opt->guisection = _("Print");
+
     /* Set description */
     module = G_define_module();
     G_add_keyword(_("database"));
```
Comment (cwhite911) [inline db/db.describe/main.c:184]:
```suggestion
    G_add_keyword(_("database"));
    G_add_keyword(_("json"));
```

**Review by cwhite911 — CHANGES_REQUESTED** [review]

File: db/db.describe/db.describe.html (hunk 2 above)
Comment (kritibirda26) [reply db/db.describe/db.describe.html:70]: Applied manually because needed to remove the existing output as well.

**Comment by kritibirda26** [comment]
@cwhite911 the pr is ready for review.

**Review by cwhite911 — APPROVED** [review]
Looks good.

**Comment by kritibirda26** [comment]
@echoix can you please merge this PR?

**Review by echoix — APPROVED** [review]

**Comment by petrasovaa** [comment]
Looks like it triggered this:

```
New defect(s) Reported-by: Coverity Scan
Showing 1 of 1 defect(s)


** CID 1559529:  Memory - illegal accesses  (UNINIT)
/db/db.describe/main.c: 86 in main()


________________________________________________________________________________________________________
*** CID 1559529:  Memory - illegal accesses  (UNINIT)
/db/db.describe/main.c: 86 in main()
80     
81         if (db_describe_table(driver, &table_name, &table) != DB_OK)
82             G_fatal_error(_("Unable to describe table <%s>"),
83                           db_get_string(&table_name));
84     
85         if (!parms.printcolnames)
>>>     CID 1559529:  Memory - illegal accesses  (UNINIT)
>>>     Using uninitialized value "root_object" when calling "print_table_definition".
86             print_table_definition(driver, table, parms.format, root_object,
87                                    cols_array);
88         else {
89             ncols = db_get_table_number_of_columns(table);
90     
91             db_init_string(&stmt);
```

**Pull request merged**